sljit_src/sljitNativeX86_common.c

```diff
@@ -2220,6 +2363,8 @@ SLJIT_API_FUNC_ATTRIBUTE sljit_s32 sljit_emit_op_src(struct sljit_compiler *comp
 	CHECK_EXTRA_REGS(src, srcw, (void)0);
 
 	switch (op) {
+	case SLJIT_FAST_RETURN_SKIP_FRAMES:
+		FAIL_IF(adjust_shadow_stack(compiler, src, srcw, 0));
```

Comment:
Maybe I wasn't clear, but I want this as a separate operation, and not part of `SLJIT_FAST_RETURN`.

Ideally you could do the following:
sljit_emit_op_src(compiler, SLJIT_SKIP_FRAMES_BEFORE_FAST_RETURN, src, srcw);
/* Do other things, e.g. set flags, since SLJIT_SKIP_FRAMES_BEFORE_FAST_RETURN may destroy them. */
sljit_emit_op_src(compiler, SLJIT_FAST_RETURN, src, srcw);

This provides quite a flexibility since users can do the shadow stack update anytime, not when the return is actually performed.

Reply:
Fixed.